nengo/spa/vocab.py

```diff
@@ -147,6 +149,17 @@ def add(self, key, p):
 
         The pointer value can be a SemanticPointer or a vector.
         """
+        if self.read_only:
+            if self.parent is None:
+                # If this vocabulary is not a subset vocabulary, throw an error
+                raise ValueError("Cannot add semantic pointer '%s' to "
+                                 "read-only vocabulary." % key)
+            else:
+                # If this vocabulary is a subset vocabulary, apply operation on
+                # parent vocab.
+                self.parent.add(key, p)
+                return
```

Comment:
I think we should get rid of the `transform_to` magic and be explicit about transforms. But that should be the scope of another PR.